neo/Network/P2P/Message.cs

```diff
@@ -105,26 +97,61 @@ public static int TryDeserialize(ByteString data, out Message msg)
             {
                 Flags = flags,
                 Command = (MessageCommand)header[1],
-                Payload = data.Slice(payloadIndex, (int)length).ToArray()
+                _payload_compressed = data.Slice(payloadIndex, (int)length).ToArray()
             };
+            msg.SetPayload();
```

Comment:
Maybe we just want to relay the message, we should set the payload when we call the property

Reply:
Fixed in https://github.com/shargon/neo/pull/9/commits/f22fa21bfce22e4396ee00b84b099ead848a448b

Reply:
We always access `Payload` whether or not we relay the message. For example, we must verify tx before relay it. So we decompress the payload when deserializing the message to check the real size of payload.